hindsight-integrations/litellm/hindsight_litellm/wrappers.py

```diff
@@ -0,0 +1,708 @@
+"""Native client wrappers for Hindsight memory integration.
+
+This module provides wrappers for native LLM client SDKs (OpenAI, Anthropic)
+that automatically integrate with Hindsight for memory injection and storage.
+
+This is an alternative to the LiteLLM callback approach, providing direct
+integration with native client libraries.
+"""
+
+import logging
+from typing import Any, Dict, List, Optional, Union
+from dataclasses import dataclass
+
+from .config import get_config, is_configured, HindsightConfig
+
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class RecallResult:
+    """A single memory recall result."""
+    text: str
+    fact_type: str
+    weight: float
+    metadata: Optional[Dict[str, Any]] = None
+
+    def __str__(self) -> str:
+        return self.text
+
+
+def recall(
+    query: str,
+    limit: int = 10,
```

Comment:
hindsight doesn't have the limit/top-k parameter by design. can we get rid of this?